### DISCUSSION TITLE
Advice with serverless deployment and Prisma

### DISCUSSION THREAD

**Discussion by sunrunner4kr-simon** [prisma/prisma#6943] [Q&A]
Hi fellow Prisma users!

I have a Next JS app deployed in serverless Vercel, and my data is stored in a standard MySQL server accessed through Prisma.

It's only a small app, maybe once an hour performing 100 upserts in an API, so I don't need the expense of a serverless MySQL/PostgreSQL

However, I **constantly** get problems in Vercel with "cannot connect to database, is it running", "too many connections, timeout". It means that the simplest of api functions are failing, and need re-running to try and get the data updated.

e.g.
```
2021-05-05T07:36:41.347Z	9ce4d3a3-cd0f-4934-bffe-259910b6c6dd	ERROR	Update listing failed:  Can't reach database server at `banditbirds.co.uk`:`3306`
Please make sure your database server is running at `banditbirds.co.uk`:`3306`.
```

Without a pool connection manager, I'm receiving ~80 JSON objects in an API call, and upserting each one to a DB record, so I'm assuming it's connecting and reconnecting 80 times. And this is why I'm getting timeouts and problems.

I've tried the following connection params: -

`pool_timeout=30?connection_limit=20`

As it's a standard web hosted MySQL server: - 

- it won't be the same region as the Vercel application.
- I can't use pgbouncer or similar service

So I'm trying to work out what options I have for solving this problem, without having to pay for the expense of a serverless PostgreSQL at $15/month and use pgbouncer

Any advice appreciated!

**Comment by janpio** [comment]
Let's see if we can help:

What does your script do? 
Is it 1 request that gets 80 JSON objects, then does the upserting?
How do you execute that logic?
How are you connecting to the database server via Prisma, and are you closing that connection when you are done?
Is it really just 1 request at a time?
Do you have connection monitoring for that database server?
